src/Apps/Artwork/Components/OtherWorks/index.tsx

```diff
@@ -1,63 +1,134 @@
+import { Join, Spacer } from "@artsy/palette"
 import { OtherWorks_artwork } from "__generated__/OtherWorks_artwork.graphql"
+import { OtherAuctionsQueryRenderer as OtherAuctions } from "Apps/Artwork/Components/OtherAuctions"
+import { Header } from "Apps/Artwork/Components/OtherWorks/Header"
+import { RelatedWorksArtworkGridRefetchContainer as RelatedWorksArtworkGrid } from "Apps/Artwork/Components/OtherWorks/RelatedWorksArtworkGrid"
+import { Mediator, withSystemContext } from "Artsy"
+import { track, useTracking } from "Artsy/Analytics"
+import * as Schema from "Artsy/Analytics/Schema"
+import ArtworkGrid from "Components/ArtworkGrid"
 import React from "react"
 import { createFragmentContainer, graphql } from "react-relay"
-import { ArtworkContextArtistFragmentContainer as ArtworkContextArtist } from "./ArtworkContexts/ArtworkContextArtist"
-import { ArtworkContextAuctionQueryRenderer as ArtworkContextAuction } from "./ArtworkContexts/ArtworkContextAuction"
-import { ArtworkContextFairFragmentContainer as ArtworkContextFair } from "./ArtworkContexts/ArtworkContextFair"
-import { ArtworkContextPartnerShowFragmentContainer as ArtworkContextPartnerShow } from "./ArtworkContexts/ArtworkContextPartnerShow"
+import { get } from "Utils/get"
 
 export interface OtherWorksContextProps {
-  /** The artworkSlug to query */
-  artworkSlug: string
-  /** Used to exclude the current work from the currently-shown work from grid */
-  artworkID: string
+  artwork: OtherWorks_artwork
+  mediator?: Mediator
+}
+
+/**
+ * Check to see if a connection's edges have a length; if false hide the grid.
+ */
+export function hideGrid(artworksConnection): boolean {
+  return Boolean(get(artworksConnection, p => !p.edges.length))
+}
+
+const populatedGrids = (grids: OtherWorks_artwork["contextGrids"]) => {
+  if (grids && grids.length > 0) {
+    return grids.filter(grid => {
+      return (
+        grid.artworks &&
+        grid.artworks.edges &&
+        grid.artworks.edges.length > 0 &&
+        grid.__typename !== "RelatedArtworkGrid"
+      )
+    })
+  }
+}
+
+const contextGridTypeToContextModule = contextGridType => {
+  switch (contextGridType) {
+    case "ArtistArtworkGrid": {
+      return Schema.ContextModule.OtherWorksByArtist
+    }
+    case "PartnerArtworkGrid": {
+      return Schema.ContextModule.OtherWorksFromGallery
+    }
+    case "AuctionArtworkGrid": {
+      return Schema.ContextModule.OtherWorksInAuction
+    }
+    case "ShowArtworkGrid": {
+      return Schema.ContextModule.OtherWorksFromShow
+    }
+  }
 }
 
+export const OtherWorks: React.FC<{
+  artwork: OtherWorks_artwork
+}> = track()(props => {
```

Comment:
this `props` parameter is typed as `any` because it's wrapped in `track` and typescript isn't clever enough to infer the type from the context.

try this instead:

```ts
export const OtherWorks = track()(
  (props: { artwork: OtherWorks_artwork } & SystemContextProps) => {
```

I noticed this because `props.mediator` was being used below but wasn't declared up here.

Reply:
Oooh, thank you!